crates/sui-metrics-proxy/src/main.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::Parser;
+use std::net::SocketAddr;
+use sui_metrics_proxy::admin::{app, set_tracing, shutdown_signal};
+use tracing::info;
+
+const GIT_REVISION: &str = {
+    if let Some(revision) = option_env!("GIT_REVISION") {
+        revision
+    } else {
+        let version = git_version::git_version!(
+            args = ["--always", "--dirty", "--exclude", "*"],
+            fallback = ""
+        );
+
+        if version.is_empty() {
+            panic!("unable to query git revision");
+        }
+        version
+    }
+};
+const VERSION: &str = const_str::concat!(env!("CARGO_PKG_VERSION"), "-", GIT_REVISION);
+
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
+#[clap(name = env!("CARGO_BIN_NAME"))]
+#[clap(version = VERSION)]
+struct Args {
+    #[clap(long, short, help = "Specify address to listen on")]
+    listen_address: SocketAddr,
+    #[clap(long, short, help = "Specify an upstream address to send to")]
+    upstream_address: SocketAddr,
```

Comment:
Do we want this to be a String/Url instead of a socketaddr?

Reply:
i don't think so, SocketAddr works for listening services, it should work for dialing also right?

Reply:
a socket addr is something like `127.0.0.1:0` where as we probably want an url so we can configure where the data is sent instead of hard coding it (as well as we probably want to be able to benefit from dns), e.g. `https://metrics.sui.io/grafana-endpoint` or something along those lines.

Reply:
yeah, you're right, doesn't work with a url.  running this locally, i was just providing the ip manually; `[::1]:8081`, for example. i'll update it to a string for the upstream. I would have hit that when i tried to start uploading to grafana.